flink-connector-mongodb/src/main/java/org/apache/flink/connector/mongodb/table/MongoFilterPushDownVisitor.java

```diff
@@ -0,0 +1,251 @@
+
+import java.math.BigDecimal;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Visitor that convert Expression to Bson filter. Return {@link Filters#empty()} if we cannot push
+ * down the filter.
+ */
+@Experimental
+public class MongoFilterPushDownVisitor extends ExpressionDefaultVisitor<BsonValue> {
+
+    public static final MongoFilterPushDownVisitor INSTANCE = new MongoFilterPushDownVisitor();
+
+    private MongoFilterPushDownVisitor() {
+        // Singleton instance.
+    }
+
+    @Override
+    public BsonDocument visit(CallExpression call) {
+        Bson filter = Filters.empty();
+        if (BuiltInFunctionDefinitions.EQUALS.equals(call.getFunctionDefinition())) {
+            filter = renderBinaryComparisonOperator("$eq", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.LESS_THAN.equals(call.getFunctionDefinition())) {
+            filter = renderBinaryComparisonOperator("$lt", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.LESS_THAN_OR_EQUAL.equals(call.getFunctionDefinition())) {
+            filter = renderBinaryComparisonOperator("$lte", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.GREATER_THAN.equals(call.getFunctionDefinition())) {
+            filter = renderBinaryComparisonOperator("$gt", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.GREATER_THAN_OR_EQUAL.equals(call.getFunctionDefinition())) {
+            filter = renderBinaryComparisonOperator("$gte", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.NOT_EQUALS.equals(call.getFunctionDefinition())) {
+            filter = renderBinaryComparisonOperator("$ne", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.IS_NULL.equals(call.getFunctionDefinition())) {
+            filter =
+                    renderUnaryComparisonOperator(
+                            "$eq", call.getResolvedChildren().get(0), BsonNull.VALUE);
+        }
+        if (BuiltInFunctionDefinitions.IS_NOT_NULL.equals(call.getFunctionDefinition())) {
+            filter =
+                    renderUnaryComparisonOperator(
+                            "$ne", call.getResolvedChildren().get(0), BsonNull.VALUE);
+        }
+        if (BuiltInFunctionDefinitions.OR.equals(call.getFunctionDefinition())) {
+            filter = renderLogicalOperator("$or", call.getResolvedChildren());
+        }
+        if (BuiltInFunctionDefinitions.AND.equals(call.getFunctionDefinition())) {
+            filter = renderLogicalOperator("$and", call.getResolvedChildren());
+        }
+        return filter.toBsonDocument();
+    }
+
+    private Bson renderBinaryComparisonOperator(
+            String operator, List<ResolvedExpression> expressions) {
+        Optional<FieldReferenceExpression> fieldReferenceExpr =
```

Comment:
MongoDB is more complicated in this situation, and it will cause full collection scanning, so it is temporarily regardless of the scene of Pushdown.
https://stackoverflow.com/questions/4442453/mongodb-query-condition-on-comparing-2-fields